WordPressAuthenticator/Signin/LoginSiteAddressViewController.swift

```diff
@@ -185,32 +188,78 @@ class LoginSiteAddressViewController: LoginViewController, NUXKeyboardResponder
         let successBlock: (WordPressComSiteInfo) -> Void = { [weak self] siteInfo in
             self?.loginFields.meta.siteInfo = siteInfo
             if WordPressAuthenticator.shared.delegate?.allowWPComLogin == false {
+                // Hey, you have to log out of your existing WP.com account before logging into another one.
                 self?.promptUserToLogoutBeforeConnectingWPComSite()
                 self?.configureViewLoading(false)
             } else {
-                self?.showSelfHostedUsernamePassword()
+                self?.configureViewLoading(false)
+                guard let strongSelf = self else {
```

Comment:
We can do `guard let self = self` in Swift 4.2

Reply:
Nice! I didn't know that.